lobsterpy/structuregraph/graph.py

```diff
@@ -0,0 +1,171 @@
+                self.path_to_icooplist = path_to_icooplist
+                self.path_to_icobilist = path_to_icobilist
+            else:
+                raise ValueError(
+                    "add_additional_data_sg is set to True."
+                    "Please provide path_to_icooplist and path_to_icobilist"
+                )
+        else:
+            self.add_additional_data_sg = add_additional_data_sg
+
+        self.path_to_poscar = path_to_poscar
+        self.path_to_charge = path_to_charge
+        self.path_to_cohpcar = path_to_cohpcar
+        self.path_to_icohplist = path_to_icohplist
+        self.path_to_madelung = path_to_madelung
+        self.which_bonds = which_bonds
+        self.cutoff_icohp = cutoff_icohp
+
+        if self.which_bonds == "all":
+            self.additional_condition = 0
+        elif self.which_bonds == "cation-anion":
+            self.additional_condition = 1
+        else:
+            raise ValueError(
+                "Only accepted values are 'all' and 'cation-anion'."
+                "Please check the input parameters of which_bonds arg"
+            )
+        self.start = start
+
+        self.sg = self.get_decorated_sg()
+
+    def get_decorated_sg(self):
+        """
+        Method to generate graph object decorated with bonding data from lobsterpy
+        Returns:
+            structure graph object
+        """
+        if self.add_additional_data_sg:
+            chemenvlobster = LobsterNeighbors(
+                are_coops=False,
+                filename_ICOHP=self.path_to_icohplist,
+                perc_strength_ICOHP=self.cutoff_icohp,
+                structure=Structure.from_file(self.path_to_poscar),
+                additional_condition=self.additional_condition,
+                filename_CHARGE=self.path_to_charge,
+                add_additional_data_sg=self.add_additional_data_sg,
+                filename_blist_sg1=self.path_to_icobilist,
+                id_blist_sg1="ICOBI",
+                filename_blist_sg2=self.path_to_icooplist,
+                id_blist_sg2="ICOOP",
+                valences_from_charges=True,
+                adapt_extremum_to_add_cond=True,
+            )
+
+        else:
+            chemenvlobster = LobsterNeighbors(
+                are_coops=False,
+                filename_ICOHP=self.path_to_icohplist,
+                perc_strength_ICOHP=self.cutoff_icohp,
+                structure=Structure.from_file(self.path_to_poscar),
+                additional_condition=self.additional_condition,
+                filename_CHARGE=self.path_to_charge,
+                add_additional_data_sg=self.add_additional_data_sg,
+                valences_from_charges=True,
+                adapt_extremum_to_add_cond=True,
+            )
+
+        # Adds Mulliken and Löwdin charges as site properties to structure object (node properties)
+        decorated_structure = Charge(self.path_to_charge).get_structure_with_charges(
```

Comment:
Isn't the charge information already in LobsterNeighbors or the Analysis object or anything else?

Reply:
Oh yes, justed looked up into get_structure_with_charges method, I infact do not need to reinitialize charge object just for that method. Thanks for the suggestion. Will implement the changes soon

Reply:
Ops, we cannot, Lobsterpy only has Mulliken charges, Loewdin Charges are not part of Analysis object

Reply:
I see! Thanks for checking!